osquery/tables/system/windows/scheduled_tasks.cpp

```diff
@@ -0,0 +1,160 @@
+    {TASK_STATE_READY, "ready"},
+    {TASK_STATE_RUNNING, "running"},
+};
+
+void enumerateTasksForFolder(std::string path, QueryData& results) {
+  void* vpService = nullptr;
+  auto ret = CoCreateInstance(CLSID_TaskScheduler,
+                              nullptr,
+                              CLSCTX_INPROC_SERVER,
+                              IID_ITaskService,
+                              &vpService);
+  if (FAILED(ret)) {
+    VLOG(1) << "Failed to create COM instance " << ret;
+    return;
+  }
+
+  auto pService = static_cast<ITaskService*>(vpService);
+  ret =
+      pService->Connect(_variant_t(), _variant_t(), _variant_t(), _variant_t());
+  if (FAILED(ret)) {
+    VLOG(1) << "Failed to connect to TaskService " << ret;
+    pService->Release();
+    return;
+  }
+
+  ITaskFolder* pRootFolder = nullptr;
+  auto widePath = stringToWstring(path.c_str());
+  ret = pService->GetFolder(BSTR(widePath.c_str()), &pRootFolder);
+  pService->Release();
+  if (FAILED(ret)) {
+    VLOG(1) << "Failed to get root task folder " << ret;
+    return;
+  }
+
+  IRegisteredTaskCollection* pTaskCollection = nullptr;
+  ret = pRootFolder->GetTasks(TASK_ENUM_HIDDEN, &pTaskCollection);
+  pRootFolder->Release();
+  if (FAILED(ret)) {
+    VLOG(1) << "Failed to get task collection for root folder " << ret;
+    return;
+  }
+
+  long numTasks = 0;
+  pTaskCollection->get_Count(&numTasks);
+  for (size_t i = 0; i < numTasks; i++) {
+    IRegisteredTask* pRegisteredTask = nullptr;
+    // Colletions are 1-based lists
+    ret = pTaskCollection->get_Item(_variant_t(i + 1), &pRegisteredTask);
+    if (FAILED(ret)) {
+      VLOG(1) << "Failed to process task";
+      continue;
+    }
+
+    Row r;
+    BSTR taskName;
+    ret = pRegisteredTask->get_Name(&taskName);
+    std::wstring wTaskName(taskName, SysStringLen(taskName));
+    r["name"] = ret == S_OK ? SQL_TEXT(wstringToString(wTaskName.c_str())) : "";
+
+    VARIANT_BOOL enabled = false;
+    pRegisteredTask->get_Enabled(&enabled);
+    r["enabled"] = enabled ? INTEGER(1) : INTEGER(0);
+
+    TASK_STATE taskState;
+    pRegisteredTask->get_State(&taskState);
+    r["state"] = kStateMap.count(taskState) > 0
+                     ? kStateMap.at(taskState)
+                     : kStateMap.at(TASK_STATE_UNKNOWN);
+
+    BSTR taskPath;
+    ret = pRegisteredTask->get_Path(&taskPath);
+    std::wstring wTaskPath(taskPath, SysStringLen(taskPath));
+    r["path"] = ret == S_OK ? SQL_TEXT(wstringToString(wTaskPath.c_str())) : "";
+
+    VARIANT_BOOL hidden = false;
+    pRegisteredTask->get_Enabled(&hidden);
+    r["hidden"] = hidden ? INTEGER(1) : INTEGER(0);
+
+    HRESULT lastTaskRun = E_FAIL;
+    pRegisteredTask->get_LastTaskResult(&lastTaskRun);
+    _com_error err(lastTaskRun);
+    r["last_run_message"] = err.ErrorMessage();
+    r["last_run_code"] = INTEGER(lastTaskRun);
+
+    DATE dRunTime;
+    pRegisteredTask->get_LastRunTime(&dRunTime);
+    COleDateTime cdt(dRunTime);
```

Comment:
Looks like our fail was due to this function's header requirement. I think I've found a way around this and will update.

It might also be good for us to, similar to the Windows Events Table, have a `time` and `datetime` column, where `time` is like the unix epoch and `datetime` is a string formated time?